### PR TITLE
feat: add telemetry to executor

### PR DESCRIPTION
**Pull request by ahkohd**
# Summary of changes

**_Summarize_** the description of your changes.

If it fixes a bug or creates a feature, link the issue.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

Release Notes:
- New Feature: Adds telemetry and Prometheus metrics tracking to the executor.
- Refactor: Modifies existing logic related to Prometheus metrics tracking.
- Chore: Minor changes such as removing unused imports, adding type annotations, and registering a listener for circuit disconnections.

> Metrics tracking and telemetry,
> Keep our system running with great efficiency.
> With types and minor tweaks,
> Our code is now at its peak.

### CODE REVIEW

File: client/packages/executor/src/prometheus.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      name: "been_out_bid",
+      help: "Number of times been out bid",
+      registers: [this.register],
+    });
+
+    this.startServer();
+  }
+
+  startServer() {
+    const port = 8080;
```
Comment (chiefbiiko) [inline client/packages/executor/src/prometheus.ts:88]: maybe not hardcode the port but it expose it through config

Comment (ahkohd) [reply client/packages/executor/src/prometheus.ts:88]: okay

Comment (ahkohd) [reply client/packages/executor/src/prometheus.ts]: Fixed, see commit [a87c786](https://github.com/t3rn/t3rn/pull/1051/commits/a87c78633419a638ae002349ee631964e61e102a).

**Review by 3h4x — APPROVED** [review]
good stuff!
left small comments. please address them before merge

**Review by chiefbiiko — APPROVED** [review]

File: client/packages/executor/src/index.ts [hunk 2]
```diff
@@ -86,6 +93,17 @@ class Instance {
     this.sdk = new Sdk(this.config.circuit.rpc, this.signer);
     // @ts-ignore
     this.circuitClient = await this.sdk.init();
+
+    this.circuitClient.on("disconnected", () => {
```
[the remark this answers is not in the document]
Comment (ahkohd) [reply client/packages/executor/src/index.ts:97]: Does this commit fix this. [3c6e4cc](https://github.com/t3rn/t3rn/pull/1051/commits/3c6e4ccbd3965c9a9cbd64bed936fbf957e4c728)

File: client/packages/executor/src/prometheus.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  startServer() {
+    const port = process.env.PROMETHEUS_PORT || 3001;
+    const server = http.createServer(async (req, res) => {
+      try {
+        if (req.url === "/metrics") {
+          const metrics = await this.register.metrics();
+          res.setHeader("Content-Type", this.register.contentType);
+          res.end(metrics);
+        } else if (req.url === "/status") {
```
[the remark this answers is not in the document]
Comment (ahkohd) [reply client/packages/executor/src/prometheus.ts]: Fixed, see to commit [9e37a1a](https://github.com/t3rn/t3rn/pull/1051/commits/9e37a1a4e787d6d0ee286de67fd181f9a5399fe4).

File: client/packages/executor/src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (3h4x) [reply client/packages/executor/src/index.ts:97]: Instance.prom.circuitDisconnects.inc({
        endpoint: this.config.circuit.rpc,
      });
      Instance.prom.circuitDisconnected.inc({
        endpoint: this.config.circuit.rpc,
      });

We only need one metric

[the remark this answers is not in the document]
Comment (ahkohd) [reply client/packages/executor/src/index.ts:97]: Fixed, refer to commit [5b238d4](https://github.com/t3rn/t3rn/pull/1051/commits/5b238d488d619f3f96a6837e5a7d4225a7dcc039) and [6b97ef2](https://github.com/t3rn/t3rn/pull/1051/commits/6b97ef24cbe9264f6f2773a6660575609db32ed8).

**Pull request merged**